api/src/test/java/org/apache/iceberg/expressions/TestAggregateEvaluator.java

```diff
@@ -0,0 +1,187 @@
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 2345)));
+
+  private static final DataFile FILE_2 =
+      new TestDataFile(
+          "file_2.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 0L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 100)));
+
+  private static final DataFile FILE_3 =
+      new TestDataFile(
+          "file_3.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), -33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 3333)));
+
+  private static final DataFile FILE_4 =
+      new TestDataFile(
+          "file_4.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 10L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), -1000)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 1000)));
+
+  private static final DataFile[] dataFiles = {FILE, FILE_2, FILE_3, FILE_4};
+
+  @Test
+  public void testIntAggregate() {
+    List<Expression> list =
+        ImmutableList.of(Expressions.count("id"), Expressions.max("id"), Expressions.min("id"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {70L, 3333, -1000};
+    assertEvaluatorResult(result, expected);
+  }
+
+  @Test
+  public void testAllNulls() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.count("all_nulls"),
+            Expressions.max("all_nulls"),
+            Expressions.min("all_nulls"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {0L, null, null};
+    assertEvaluatorResult(result, expected);
+  }
+
+  @Test
+  public void testSomeNulls() {
```

Comment:
Isn't this just testing the fact that the stats are missing and this can't be calculated?

I think it would be more interesting to test valid count(*) and count(some_nulls) cases where the stats exist.

Reply:
> Isn't this just testing the fact that the stats are missing and this can't be calculated?

Right.

`count(*)` has values for all the 3 files. It's `90L`

`count("some_nulls")` has value in the first FILE, the value is `50L(value counts) - 40L (null value counts)`.
It doesn't have value in the second FILE (null value counts missing), so this Aggregator is invalid and we stop looking this Aggregator. So the value of count("some_nulls") is `null`.

`max(("some_nulls")` doesn't have value in the first FILE (hasBound == false and boundAllNull == false), so this Aggregator is invalid and we will stop looking this Aggregator. So the value of `max("some_nulls")` is `null.`

`min(("some_nulls")` doesn't have value in the first FILE (hasBound == false and boundAllNull == false), so this Aggregator is invalid and we will stop looking this Aggregator. So the value of `min("some_nulls")` is `null`.